### DISCUSSION TITLE
Ionic App Capacitor error: namespace not specified in gradle.build

### DISCUSSION THREAD

**Discussion by DongJeff** [ionic-team/capacitor#6506] [Q&A]
This is an Ionic Hybrid App using Capacitor. Here is my build.gradle, namespace is clearly specified, however when attempting to build, rebuild, or clean build this error returns: 
Namespace not specified. Please specify a namespace in the module's build.gradle file like so:

android {
    namespace 'com.example.namespace'
}

If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

Here is gradle.build:

apply plugin: 'com.android.application'

android {
    namespace 'com.ionicthemes.ionic5fullapp'
    testNamespace 'com.ionicthemes.ionic5testapp'
    compileSdkVersion rootProject.ext.compileSdkVersion
    defaultConfig {
        applicationId "com.ionicthemes.ionic5fullapp"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

repositories {
    flatDir{
        dirs '../capacitor-cordova-android-plugins/src/main/libs', 'libs'
    }
}

dependencies {
    implementation "androidx.core:core-splashscreen:$coreSplashScreenVersion"
    implementation "androidx.coordinatorlayout:coordinatorlayout:$androidxCoordinatorLayoutVersion"
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
    implementation project(':capacitor-android')
    testImplementation "junit:junit:$junitVersion"
    androidTestImplementation "androidx.test.ext:junit:$androidxJunitVersion"
    androidTestImplementation "androidx.test.espresso:espresso-core:$androidxEspressoCoreVersion"
    implementation project(':capacitor-cordova-android-plugins')
    implementation 'com.facebook.android:facebook-android-sdk:5.15.3'
}

apply from: 'capacitor.build.gradle'

try {
    def servicesJSON = file('google-services.json')
    if (servicesJSON.text) {
        apply plugin: 'com.google.gms.google-services'
    }
} catch(Exception e) {
    logger.warn("google-services.json not found, google-services plugin not applied. Push Notifications won't work")
}

**Comment by jcesarmobile** [comment]
Looks like you upgraded to Gradle 8, Gradle 8 is not compatible with Capacitor 4, will be compatible with Capacitor 5.
You should downgrade Gradle until Capacitor 5 is ready, or you can start using Capacitor 5 already (but it's still in Beta)
Check the [Capacitor 5 upgrade guide ](https://capacitorjs.com/docs/next/updating/5-0), but where it says `npm i -D @capacitor/cli@latest` do `npm i -D @capacitor/cli@next` instead.
